src/deploy/functions/prepareFunctionsUpload.ts

```diff
@@ -90,10 +90,12 @@ async function packageSource(options: Options, sourceDir: string, configValues:
         mode: file.mode,
       });
     });
-    archive.append(JSON.stringify(configValues, null, 2), {
-      name: CONFIG_DEST_FILE,
-      mode: 420 /* 0o644 */,
-    });
+    if (typeof configValues !== "undefined") {
```

Comment:
q: Is this equivalent to:

```js
if (configValues !== undefined) {
...
```

Is there reason to prefer one over another?

Reply:
Though it doesn't seem to work in new versions of Node, I was raised in a world where `undefined` was a variable and not a literal, so you were always supposed to check the type of a value and not compare it to the variable `undefined`.